src/org/openvv/js/OVVAsset.js

```diff
@@ -540,6 +469,124 @@ OVVCheck.BEACON = 'beacon';
 */
 OVVCheck.GEOMETRY = 'geometry';
 
+function OVVBrowser(userAgent)
+{
+
+    var browserIDEnum = {
+        MSIE: 1,
+        Firefox: 2,
+        Chrome: 3,
+        Opera: 4,
+        safari: 5
+    };
+
+    /**
+     * Returns an object that contains the browser name, version, id and os if applicable
+     * @param {String} ua userAgent
+     */
+    function getBrowserDetailsByUserAgent(ua, t) {
+
+        var getData = function () {
+            var data = { ID: 0, name: '', version: '' };
+            var dataString = ua;
+            for (var i = 0; i < dataBrowsers.length; i++) {
+                // Fill Browser ID
+                if (dataString.match(new RegExp(dataBrowsers[i].brRegex)) != null) {
+                    data.ID = dataBrowsers[i].id;
+                    data.name = dataBrowsers[i].name;
+                    if (dataBrowsers[i].verRegex == null) {
+                        break;
+                    }
+                    //Fill Browser Version
+                    var brverRes = dataString.match(new RegExp(dataBrowsers[i].verRegex + '[0-9]*'));
+                    if (brverRes != null) {
+                        var replaceStr = brverRes[0].match(new RegExp(dataBrowsers[i].verRegex));
+                        data.version = brverRes[0].replace(replaceStr[0], '');
+                    }
+                    var brOSRes = dataString.match(new RegExp(winOSRegex + '[0-9\\.]*'));
+                    if (brOSRes != null) {
+                        data.os = brOSRes[0];
+                    }
+                    break;
+                }
+            }
+            return data;
+        };
+
+        var winOSRegex = '(Windows NT )';
+        var dataBrowsers = [{
+            id: 4,
+            name: 'Opera',
+            brRegex: 'OPR|Opera',
+            verRegex: '(OPR\/|Version\/)'
+        }, {
+            id: 1,
+            name: 'MSIE',
+            brRegex: 'MSIE|Trident/7.*rv:11|rv:11.*Trident/7',
+            verRegex: '(MSIE |rv:)'
+        }, {
+            id: 2,
+            name: 'Firefox',
+            brRegex: 'Firefox',
+            verRegex: 'Firefox\/'
+        }, {
+            id: 3,
+            name: 'Chrome',
+            brRegex: 'Chrome',
+            verRegex: 'Chrome\/'
+        }, {
+            id: 5,
+            name: 'Safari',
+            brRegex: 'Safari|(OS |OS X )[0-9].*AppleWebKit',
+            verRegex: 'Version\/'
+        }
+        ];
+
+        return getData();
+    }
+
+    /**
+     * browser:
+     *	{
+    *		ID: ,
+    *	  	name: '',
+    *	  	version: '',
+    *	    os: ''
+    *	};
+     */
+    var  browser = getBrowserDetailsByUserAgent(userAgent);
+
+    this.getBrowser = function()
+    {
+        return browser;
+    }
+
+    this.getBrowserIDEnum = function()
+    {
+        return browserIDEnum;
+    }
+}
+
+function OVVBeaconSupportCheck()
+{
+    var ovvBrowser = new OVVBrowser($ovv.userAgent);
```

Comment:
Why having another instance of OVVBrowser? You created it before, can't we reuse it?